React/Modules/RCTUIManager.m

```diff
@@ -757,6 +757,20 @@ - (void)_manageChildren:(NSNumber *)containerReactTag
   }
 }
 
+- (void)endEditingForShadowView:(NSNumber *)reactTag manager:(RCTUIManager *)uiManager viewRegistry:(RCTSparseArray *)viewRegistry
+{
+  RCTShadowView *shadowView = uiManager.shadowViewRegistry[reactTag];
+
+  for(RCTShadowView *subview in [shadowView reactSubviews]) {
+    if([[subview reactSubviews] count] > 0) {
+      [uiManager endEditingForShadowView:subview.reactTag manager:uiManager viewRegistry:viewRegistry];
+    } else {
+      UIView *view = viewRegistry[subview.reactTag];
+      [view resignFirstResponder];
+    }
+  }
+}
```

Comment:
You should be able to do something like `[subview endEditing:YES]` instead of doing the complete traversal. Also can remove the uiManager parameter since self == uiManager.

```objc
UIView *view = uiManager.viewRegistry[reactTag];
if (view) {
  [view endEditing:YES];
} else {
  // Recurse through the shadow hierarchy when the shadow view has no backing UIView
  for (RCTShadowView *subview in [shadowView reactSubviews]) {
    [self endEditingForShadowView:subview.reactTag viewRegistry:viewRegistry];
  }
}
```